File: src/NovelRT.Java/src/main/java/com/github/novelrt/event/SceneConstructionRequestedEvent.java
package com.github.novelrt.event;

public class SceneConstructionRequestedEvent extends Event<SceneConstructionRequestedListener> {
  public SceneConstructionRequestedEvent(long handle) {
    super(handle);
  }

  @Override
  protected native void addSubscription(SceneConstructionRequestedListener listener);

  @Override
  protected native void removeSubscription(SceneConstructionRequestedListener listener);
}

File: src/NovelRT.Java/core/src/main/kotlin/com/github/novelrt/event/SceneConstructionRequestedEvent.kt
package com.github.novelrt.event

import com.github.novelrt.internal.handle.Handle
import com.github.novelrt.codegeneration.annotations.GenerateNativeType

class SceneConstructionRequestedEvent
@GenerateNativeType private constructor(handle: Handle) : Event<SceneConstructionRequestedListener>(handle) {
  external override fun addSubscription(listener: SceneConstructionRequestedListener, hash: Int)
  external override fun removeSubscription(listener: SceneConstructionRequestedListener, hash: Int)
}
